File: basics/14-duplicate-elements/Solution.cpp
#include <vector>
#include <unordered_set>

std::vector<int> findDuplicateElements(const std::vector<int>& arr) {
    std::unordered_set<int> seen, duplicates;
    for (int num : arr) {
        if (seen.count(num)) duplicates.insert(num);
        else seen.insert(num);
    }
    return std::vector<int>(duplicates.begin(), duplicates.end());
}

File: basics/14-duplicate-elements/Solution.java
import java.util.*;

class Solution {
    public List<Integer> findDuplicateElements(int[] arr) {
        Set<Integer> seen = new HashSet<>();
        Set<Integer> duplicates = new HashSet<>();
        for (int num : arr) {
            if (seen.contains(num)) duplicates.add(num);
            else seen.add(num);
        }
        return new ArrayList<>(duplicates);
    }
}
